File: app/src/main/java/com/snatik/matches/model/BoardArrangement.java
package com.snatik.matches.model;

import android.graphics.Bitmap;
import android.support.annotation.NonNull;
import android.util.SparseArray;

import com.snatik.matches.common.Shared;
import com.snatik.matches.themes.Themes;
import com.snatik.matches.utils.Utils;

/**
 * Before game starts, engine build new board
 *
 * @author sromku
 */
public class BoardArrangement {

    /**
     * Maps a placement id to a tile image.
     */
    public SparseArray<String> tileUrls;

    /**
     * A list of all currently placed cards mapped to their unique placement IDs.
     */
    public SparseArray<Card> cards;

    /**
     * @param id The id is the number between 0 and number of possible tiles of
     *           this theme
     * @return The Bitmap of the tile
     */
    public Bitmap getTileBitmap(int id, int size) {
        String string = tileUrls.get(id);
        if (string.contains(Themes.URI_DRAWABLE)) {
            String drawableResourceName = string.substring(Themes.URI_DRAWABLE.length());
            int drawableResourceId = Shared.context.getResources().getIdentifier(drawableResourceName, "drawable", Shared.context.getPackageName());
            Bitmap bitmap = Utils.scaleDown(drawableResourceId, size, size);
            return Utils.crop(bitmap, size, size);
        }
        return null;
    }

    public boolean isPair(@NonNull Card card1, @NonNull Card card2) {
        return card1.getPairId() == card2.getPairId();
    }

}

File: app/src/main/java/de/rafaelmiranda/memory/model/BoardArrangement.kt
package de.rafaelmiranda.memory.model

import android.util.SparseArray
import de.rafaelmiranda.memory.common.Shared
import de.rafaelmiranda.memory.types.Types

/**
 * Before game starts, engine build new board
 *
 * @author sromku
 */
class BoardArrangement {

    /**
     * Maps a placement id to a tile image.
     */
    var tileUrls: SparseArray<String>? = null

    /**
     * A list of all currently placed cards mapped to their unique placement IDs.
     */
    var cards: SparseArray<Card>? = null

    /**
     * @param id The id is the number between 0 and number of possible tiles of
     * this gameType
     * @return The Bitmap of the tile
     */
    fun getTileResId(id: Int): Int {
        val string = tileUrls!!.get(id)
        val drawableResourceName = string.substring(Types.URI_DRAWABLE.length)
        return Shared
                .context
                .resources
                .getIdentifier(drawableResourceName,
                        "drawable", Shared.context.packageName)
        // val bitmap = Utils.scaleDown(drawableResourceId, size, size)
        // return Utils.crop(bitmap, size, size)
    }

    fun isPair(card1: Card, card2: Card): Boolean {
        return card1.pairId == card2.pairId
    }

}
